### PR TITLE
[CMSDS-3512] Allow text overrides of screen reader content for the Badge component

### PR DESCRIPTION
**Pull request by jack-ryan-nava-pbc**
## Summary

- Overview of changes
- [Ticket](https://jira.cms.gov/browse/CMSDS-3512)

## How to test

1. Run specs locally
2. Try it out in storybook! The text should change within the `span.ds-u-visibility--screen-reader` element from whatever the variation is to the override text you provide via the control. It should go back to the variation provided text when the prop is undefined or just a " ".

## Checklist

- [X] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [X] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [X] Selected appropriate `Impacts`, multiple can be selected.
- [X] Selected appropriate release milestone

### CODE REVIEW

File: packages/design-system/src/components/Badge/Badge.test.tsx [hunk 1]
```diff
@@ -32,4 +32,17 @@ describe('Badge', () => {
...
+    const srText = screen.getByText('barbaz', { exact: false });
+    expect(srText).toBeInTheDocument();
+  });
+
+  it('should not override the screen reader text when the prop is just a space', () => {
+    renderBadge({ variation: 'success', screenReaderText: ' ' });
+    // Note 'Success' is the value provided by the variation, not the screen reader text
+    const srText = screen.getByText('Success', { exact: false });
+    expect(srText).toBeInTheDocument();
+  });
```
Comment (jcq) [inline packages/design-system/src/components/Badge/Badge.test.tsx]: This is the only bit where I have a pause... I'm not certain of how likely it might be, but I could at least imagine a scenario where the content of the badge is sufficient and one might not want any extra text read out to screen readers. With this implementation, there's no way to override it entirely (both empty string and `undefined` will result in the default being read out).

Likely not a huge deal, but just wanted to flag

Comment (jack-ryan-nava-pbc) [reply packages/design-system/src/components/Badge/Badge.test.tsx]: @jcq Good callout! I've just pushed up some changes that introduce the option to hide this field completely via a separate prop. Take a look and let me know if those changes address your concern!

Comment (jcq) [reply packages/design-system/src/components/Badge/Badge.test.tsx]: @jack-ryan-nava-pbc Yup yup, that seems like a good option to account for that edge case. TY!

**Review by kim-cmsds — APPROVED** [review]
Adding both props LGTM ✨
